### DISCUSSION TITLE
Wizard next step helper for tests

### DISCUSSION THREAD

**Discussion by eduPHP** [filamentphp/filament#11622] [Ideas]
I just made a macro for calling the next step on a wizard form while testing a component, I wish I could submit it as a feature, is it of any use @danharrin?

### Macro
```php
\Livewire\Features\SupportTesting\Testable::macro('callNextStep', function () {
    return $this->call('dispatchFormEvent', 'wizard::nextStep', 'data', 0);
});
```

### Usage:
```php
$component->fillForm([
    'name' => null,
])->callNextStep()->assertHasFormErrors(['name']);
```

**Comment by danharrin** [comment]
Should probably be called `nextFormWizardStep()`, and have params for the `$currentStep` and `$form` and `$statePath` instead of hardcoding those into the event params

**Reply by ardzz** [reply]
in which object i can call the `nextFormWizardStep` method?

**Reply by eduPHP** [reply]
> Should probably be called `nextFormWizardStep()`, and have params for the `$currentStep` and `$form` and `$statePath` instead of hardcoding those into the event params

Just finished a initial version, the state path can be obtained from the form object itself. So, at least for resources a `$statePath` param seems unnecessary. I will dig a bit more to see if it is applicable to forms outside a resource.

Test:
```php
it('moves to next wizard step', function () {
    livewire(CreatePostUsingWizard::class) // created just for this test
        ->nextFormWizardStep()
        ->assertHasFormErrors(['title'])
        ->assertHasNoFormErrors(['content'])
        ->nextFormWizardStep(currentStep: 1)
        ->assertHasFormErrors(['content'])
        ->assertHasNoFormErrors(['title']);
});
```
   PASS  Tests\src\Panels\Resources\Pages\CreateRecordUsingWizardTest
  ✓ it moves to next wizard step                                         0.26s

  Tests:    1 passed (11 assertions)
  Duration: 0.39s

**Comment by mokhosh** [comment]
@eduPHP This is neat. @danharrin Has this been implemented by any chance? I couldn't find it in the docs.

**Reply by eduPHP** [reply]
> @eduPHP This is neat. @danharrin Has this been implemented by any chance? I couldn't find it in the docs.

I didn't submit a merge request for this yet, I will be sure to take some time today to do so!

**Comment by eduPHP** [comment]
@danharrin are tests required for this kind of helper?

**Reply by danharrin** [reply]
If you're introducing a testing helper than yes its a good idea to use that helper in at least one test case

**Comment by eduPHP** [comment]
Merge request ready 🥳

Thanks to everyone that showed interest on this 🤗

https://github.com/filamentphp/filament/issues/13278

**Comment by eduPHP** [comment]
Merged!
